runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -228,85 +540,352 @@ func s:Evaluate(range, arg)
     let expr = @v
     call setpos('.', pos)
     call setreg('v', reg, regt)
+    let s:evalFromBalloonExpr = 1
   else
     let expr = expand('<cexpr>')
+    let s:evalFromBalloonExpr = 1
   endif
-  call term_sendkeys(s:commbuf, '-data-evaluate-expression "' . expr . "\"\r")
-  let s:evalexpr = expr
+  let s:ignoreEvalError = 0
+  call s:SendEval(expr)
 endfunc
 
+let s:ignoreEvalError = 0
+let s:evalFromBalloonExpr = 0
+let s:evalFromBalloonExprResult = ''
+
 " Handle the result of data-evaluate-expression
 func s:HandleEvaluate(msg)
-  echomsg '"' . s:evalexpr . '": ' . substitute(a:msg, '.*value="\(.*\)"', '\1', '')
+  let value = substitute(a:msg, '.*value="\(.*\)"', '\1', '')
+  let value = substitute(value, '\\"', '"', 'g')
+  let value = substitute(value, '
', '\1', '')
+  if s:evalFromBalloonExpr
+    if s:evalFromBalloonExprResult == ''
+      let s:evalFromBalloonExprResult = s:evalexpr . ': ' . value
+    else
+      let s:evalFromBalloonExprResult .= ' = ' . value
+    endif
+    call s:OpenHoverPreview([s:evalFromBalloonExprResult], v:null)
+  else
+    echomsg '"' . s:evalexpr . '": ' . value
+  endif
+
+  if s:evalexpr[0] != '*' && value =~ '^0x' && value != '0x0' && value !~ '"$'
+    " Looks like a pointer, also display what it points to.
+    let s:ignoreEvalError = 1
+    call s:SendEval('*' . s:evalexpr)
+  else
+    let s:evalFromBalloonExprResult = ''
+  endif
 endfunc
 
+function! s:ShouldUseFloatWindow() abort
+  if has('nvim_open_win') && exists('g:termdebug_useFloatingHover') && (g:termdebug_useFloatingHover == 1)
+    return v:true
+  else
+    return v:false
+  endif
+endfunction
+
+function! s:CloseFloatingHoverOnCursorMove(win_id, opened) abort
+  if getpos('.') == a:opened
+    " Just after opening floating window, CursorMoved event is run.
+    " To avoid closing floating window immediately, check the cursor
+    " was really moved
+    return
+  endif
+  autocmd! plugin-LC-neovim-close-hover
+  let winnr = win_id2win(a:win_id)
+  if winnr == 0
+    return
+  endif
+  execute winnr . 'wincmd c'
+endfunction
+
+function! s:CloseFloatingHoverOnBufEnter(win_id, bufnr) abort
+    let winnr = win_id2win(a:win_id)
+    if winnr == 0
+        " Float window was already closed
+        autocmd! plugin-LC-neovim-close-hover
+        return
+    endif
+    if winnr == winnr()
+        " Cursor is moving into floating window. Do not close it
+        return
+    endif
+    if bufnr('%') == a:bufnr
+        " When current buffer opened hover window, it's not another buffer. Skipped
+        return
+    endif
+    autocmd! plugin-LC-neovim-close-hover
+    execute winnr . 'wincmd c'
```

Comment:
call `nvim_win_close(a:win_id)` is more robust, it works even if the user switched tab.

Reply:
thanks, this makes me think that I should also change the name of that autocmd so that it does not clash with languageclient-neovim...